eos/utils/kmatrix-impl.hh

```diff
@@ -0,0 +1,213 @@
+        _Khat = nullptr;
+    }
+
+
+    //Return row n° rowindex of the T matrix defined as T = (1-i*rho*K)^(-1)*K
+    template <unsigned nchannels_, unsigned nresonances_>
+    std::array<complex<double>, nchannels_>
+    KMatrix<nchannels_, nresonances_>::tmatrix_row(unsigned rowindex, double s) const
+    {
+
+        std::array<complex<double>, nchannels_> tmatrixrow;
+        auto channels = this->_channels;
+        auto resonances = this->_resonances;
+        auto bkgcst = this->_bkgcst;
+
+
+        ///////////////////
+        //1. Fill tmp1 = rho matrix
+        ///////////////////
+        gsl_matrix_complex_set_zero(_tmp_1);
+
+        for (size_t i = 0 ; i < nchannels_ ; ++i)
+            {
+            complex<double> rhoentry = channels[i]->rho(s);
+            gsl_matrix_complex_set(_tmp_1,  i,  i, gsl_complex_rect(rhoentry.real(), rhoentry.imag()));
+            }
+
+        ///////////////////
+        // 2. Fill Khat
+        ///////////////////
+        for (size_t i = 0 ; i < nchannels_ ; ++i)
+            {
+            for (size_t j = 0 ; j < nchannels_ ; ++j)
+                {
+                complex<double> entry = complex<double>(bkgcst[i][j].evaluate(), 0.0);
+
+                for (size_t a = 0 ; a < nresonances_ ; ++a)
+                    {
+
+                    double mres = resonances[a]->_m;
+
+                    Parameter g0rci = channels[i]->_g0s[a];
+                    Parameter g0rcj = channels[j]->_g0s[a];
+
+
+                    // Avoid the pole by adding a small imaginary part
+                    complex<double> mres2 = { mres * mres, 1.0e-7 };
+                    entry += g0rci * g0rcj / (mres2 - s);
+
+                    }
+
+                gsl_matrix_complex_set(_Khat, i, j, gsl_complex_rect(entry.real(), entry.imag()));
+                }
+            }
+
+        ///////////////////
+        // 3. Compute That
+        ///////////////////
+        static const gsl_complex one  = gsl_complex_rect(1.0, 0.0);
+        static const gsl_complex minusi  = gsl_complex_rect(0.0, -1.0);
+        static const gsl_complex zero = gsl_complex_rect(0.0, 0.0);
+
+        // 3a. multiply -i*rho with Khat from the right
+        // -> tmp_2 = -i * tmp_1 * Khat + zero * tmp_2 ; no transpose anywhere
+        gsl_blas_zgemm(CblasNoTrans, CblasNoTrans, minusi, _tmp_1, _Khat, zero, _tmp_2);
+
+        // 3b. add 1.0 to the diagonal elements of tmp_2
+        for (unsigned i = 0 ; i < nchannels_ ; ++i)
+            {
+            gsl_complex value = gsl_matrix_complex_get(_tmp_2, i, i);
+            gsl_matrix_complex_set(_tmp_2, i, i, gsl_complex_add_real(value, 1.0));
+            }
+
+        // 3c. invert tmp_2 and assign the result to tmp_1
+        int signum = 0;
+        gsl_permutation_init(_perm);
+        gsl_linalg_complex_LU_decomp(_tmp_2, _perm, &signum);
+        gsl_linalg_complex_LU_invert(_tmp_2, _perm, _tmp_1);
+
+        // 3d. calculate That = Khat / (1 - i rho * Khat) = Khat * tmp_1
```

Comment:
Please indicate in the comment that on the center and right-hand side of the equation we have ```zero * That``` as an additional term.